### PR TITLE
Add list of mapped tasks to key data item

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

Moves the manipulation of spec tags and metadata to reconstruction while result in less code and I believe is more clear.

## How I Tested These Changes

BK

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by schrockn** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24788?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24788** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24788?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24787** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24787?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24770?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24769?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24766** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24766?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24765** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24765?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @schrockn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py [hunk 1]
```diff
@@ -357,6 +358,7 @@ def file_token(self) -> str:
         return self.metadata["file_token"]
 
 
+@whitelist_for_serdes
```
Comment (dpeng817) [inline examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py:361]: this does concern me ever so slightly, means we might need to start constraining our field retrieval

Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/core/airflow_instance.py:361]: also, this should move into the serialized classes file

**Comment by schrockn** [comment]
### Merge activity

* **Sep 26, 8:01 PM EDT**: @schrockn started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/24788).

**Pull request merged**